### PR TITLE
chore: clear old txs

### PR DESCRIPTION
**Pull request by namgold**

### CODE REVIEW

File: src/state/transactions/reducer.ts [hunk 1]
```diff
@@ -12,16 +13,23 @@ import {
...
+  [chainId in ChainId]?: GroupedTxsByHash | undefined
 }
 
 const initialState: TransactionState = {}
 
+const clearOldTransactions = (transactions: GroupedTxsByHash | undefined): GroupedTxsByHash | undefined => {
+  if (!transactions) return undefined
+  const chainTxs = Object.values(transactions ?? {}).filter(Boolean) as TransactionDetails[][]
+  chainTxs.sort((a, b) => a[0].addedTime - b[0].addedTime)
+  const slicedChainTxs = chainTxs.slice(-10).filter(tx => tx[0].addedTime > Date.now() - 7 * 24 * 60 * 60 * 1000)
```
Comment (nguyenhoaidanh) [inline src/state/transactions/reducer.ts:28]:
```suggestion
  const slicedChainTxs = chainTxs.filter(tx => tx[0].addedTime > Date.now() - 7 * 24 * 60 * 60 * 1000).slice(-10)
```

Comment (namgold) [reply src/state/transactions/reducer.ts:28]: slice first then filter would have better performance

Comment (nguyenhoaidanh) [reply src/state/transactions/reducer.ts:28]: what is your expect?
if u want to show 10 txs. it should filter first and then slice

Comment (namgold) [reply src/state/transactions/reducer.ts:28]: I don't expect to be exact 10 latest txs. My expect is just maximum 10 latest txs since a week ago

**Review by nguyenhoaidanh — APPROVED** [review]
dont forget to test it by yourself

File: src/state/transactions/reducer.ts (hunk 1 above)
Comment (namgold) [reply src/state/transactions/reducer.ts:28]: If there is more than 10 txs in a week, then results would be same, both return 10 txs

If there is less than 10 txs in this week (I assume 7), mine slice took 3 additional txs before this week, then filter cut them to result 7 txs.
Your filter cut to 7 txs, then slice has nothing to do => result also 7
=> results are same too

**Pull request merged**